src/org/rust/lang/core/psi/util/RustModExtensions.kt

```diff
@@ -0,0 +1,61 @@
+package org.rust.lang.core.psi.util
+
+import com.intellij.psi.PsiDirectory
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RustCompositeElement
+import org.rust.lang.core.psi.RustCrateItem
+import org.rust.lang.core.psi.RustModItem
+
+
+private val RustModItem.modDir: PsiDirectory?
+    get() {
+        val parent = parentModInFile ?: return containingFile.parent
+
+        // FIXME(matklad): mod name should be not null, should we rearange our rules for file/mod?
+        val name = this.name ?: return null
+
+        return parent.modDir?.findSubdirectory(name)
+    }
+
+
+val RustCompositeElement.parentModInFile: RustModItem?
+    get() = parentOfType<RustModItem>()
```

Comment:
>parentModInFile sounds bulky, why not containingMod?

I didn't use `parentMod` because in general it can be in other file, so I named it `parentModInFile`. `containingMod` is indeed better.